### PR TITLE
kernel: bump 5.4 to 5.4.128

### PR DESCRIPTION
**Pull request by graysky2**
Removed upstreamed:
  mvebu/patches-5.4/002-PCI-aardvark-Don-t-rely-on-jiffies-while-holding-spi.patch

All other patches automatically rebased.

Build system: x86_64
Build-tested: ipq806x/R7800
Run-tested: ipq806x/R7800

No dmesg regressions, everything functional

Signed-off-by: John Audia <graysky@archlinux.us>

### CODE REVIEW

**Comment by zuzia-dev** [comment]
Build system: x86_64
Build-tested: lantiq/xrx200/Easybox 904 xDSL 
Run-tested: lantiq/xrx200/Easybox 904 xDSL 

OK.

File: target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch [hunk 1]
```diff
@@ -17,16 +17,18 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/kernel/cgroup/cgroup.c
 +++ b/kernel/cgroup/cgroup.c
-@@ -5722,6 +5722,8 @@ int __init cgroup_init_early(void)
+@@ -5721,6 +5721,10 @@ int __init cgroup_init_early(void)
+ 	return 0;
  }
  
- static u16 cgroup_disable_mask __initdata;
++static u16 cgroup_disable_mask __initdata;
```
Comment (hauke) [inline target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch:24]: I am not sure if this patch still works as before, have a look at this change: 
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=45e1ba40837ac2f6f4d4716bddb8d44bd7e4a251

```cgroup_disable_mask``` is not handled in the upstream code any more.

Comment (hauke) [reply target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch:24]: I think  using this patch from kernel 5.10 is a better idea:
https://github.com/raspberrypi/linux/commit/2b13c54592135b6fab269517ed687fa9f80bf8e5

**Comment by hauke** [comment]
Please add this to the generic config for kernel 5.4.126:
```
# CONFIG_NVME_TCP is not set
```

**Review by graysky2 — COMMENTED** [review]
> Please add this to the generic config for kernel 5.4.126:
> CONFIG_NVME_TCP is not set

@hauke - I'm wondering if we should make this a separate PR like https://github.com/openwrt/openwrt/pull/3143 or is it fine here?

File: target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch [hunk 2]
```diff
@@ -17,16 +17,18 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/kernel/cgroup/cgroup.c
 +++ b/kernel/cgroup/cgroup.c
-@@ -5722,6 +5722,8 @@ int __init cgroup_init_early(void)
+@@ -5721,6 +5721,10 @@ int __init cgroup_init_early(void)
```
[the remark this answers is not in the document]
Comment (pelwell) [reply target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch]: Reading a diff of a patch (itself a diff) is a bit meta, but it looks correct.

[the remark this answers is not in the document]
Comment (graysky2) [reply target/linux/bcm27xx/patches-5.4/950-0089-cgroup-Disable-cgroup-memory-by-default.patch]: Thanks, Phil.  We ended up just pulling https://github.com/raspberrypi/linux/commit/2b13c54592135b6fab269517ed687fa9f80bf8e5 into the actual patch.  I agree that reading the diffs of a patch is not trivial.

**Comment by hauke** [comment]
The new symbol ```CONFIG_NVME_TCP``` shows up because  of the kernel update, it shows be integrated into the commit which dos the upgrade.

**Comment by hauke** [comment]
Thank you for the patches, I applied them to master.

**Pull request closed without merging**